arch/arm/Kconfig

```diff
@@ -908,6 +908,11 @@ config ARM_FP_ABI_SOFT
 	---help---
 		Pass float value via integer register (-mfloat-abi=softfp) 
 
+config ARM_FP_NEON
```

Comment:
why not directly use ARM_HAVE_NEON in Makefile?

Reply:
I think because MCU can have neon, but FP usage can be disabled

Reply:
This config can't work, FPU and NEON share the same register set. OS save and restore this register set on when FPU is enabled.

Reply:
<img width="508" alt="image" src="https://user-images.githubusercontent.com/9030616/156487915-9a0f6c8c-8b02-41d5-b981-625fd88a86a9.png">
@xiaoxiang781216 Should we use these recommanded option while NEON exists?

Reply:
Yes, I think. fp16/fp32 can get form CONFIG_ARM_HAVE_FPU_D32

Reply:
CONFIG_ARM_HAVE_FPU_D32 meas there are 32 float point registers in FPU instead of 32-bit float point.

Reply:
```
D32 -> 32 double float register -> 64 single float register
       ~~
F32 -> 16 double float register -> 32 single float register
                                   ~~
```

Reply:
Got it